airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/indexer.py

```diff
@@ -170,10 +172,22 @@ def index(self, document_chunks: Iterable[Any], namespace: str, stream: str):
             cortex_processor.process_airbyte_messages(airbyte_messages, self.get_write_strategy(stream))
 
     def delete(self, delete_ids: list[str], namespace: str, stream: str):
-        # delete is generally used when we use full refresh/overwrite strategy.
-        # PyAirbyte's sync will take care of overwriting the records. Hence, we don't need to do anything here.
+        # this delete is specific to vector stores, hence not implemented here
         pass
 
+    def pre_sync(self, catalog: ConfiguredAirbyteCatalog) -> None:
```

Comment:
new method - meant to be implemented for vector dbs - deletes records beforehand for overwrite.